modules/agar/src/test/ts/browser/api/ClipboardTest.ts

```diff
@@ -1,103 +1,162 @@
+  };
+
+  before(() => {
+    pastebinState.set(SugarElement.fromHtml('<div class="pastebin"></div>'));
+    pastebinState.on((pastebin) => {
+      Insert.append(SugarBody.body(), pastebin);
+
+      const cutUnbinder = DomEvent.bind(pastebin, 'cut', (evt) => {
+        const dataTransfer = evt.raw.clipboardData;
+        dataTransfer.clearData();
+        dataTransfer.setData('text/plain', 'cut-data');
+      });
+
+      const copyUnbinder = DomEvent.bind(pastebin, 'copy', (evt) => {
+        const dataTransfer = evt.raw.clipboardData;
+        dataTransfer.clearData();
+        dataTransfer.setData('text/plain', 'copy-data');
+      });
+
+      const pasteUnbinder = DomEvent.bind(pastebin, 'paste', (evt) => {
+        const dataTransfer = evt.raw.clipboardData;
+        pasteState.set(dataTransfer);
+      });
+
+      unbinderState.set(() => {
+        cutUnbinder.unbind();
+        copyUnbinder.unbind();
+        pasteUnbinder.unbind();
+      });
+    });
+  });
+
+  after(() => {
+    pastebinState.on(Remove.remove);
+    pastebinState.clear();
+    unbinderState.on((unbind) => unbind());
+    unbinderState.clear();
+  });
+
+  it('Paste text and html items', () => {
+    const pastebin = pastebinState.get().getOrDie('Could not get pastebin from state');
+
+    pasteItems(pastebin, {
+      'text/plain': 'Hello world!',
+      'text/html': '<b>Hello world!</b>'
+    });
+
+    const dataTransfer = pasteState.get().getOrDie('Could not get dataTransfer from state');
+
+    assert.equal(dataTransfer.items.length, 2);
+    assertStringItem(dataTransfer.items[0], { type: 'text/plain', data: 'Hello world!' });
+    assertStringItem(dataTransfer.items[1], { type: 'text/html', data: '<b>Hello world!</b>' });
+  });
+
+  it('Paste text and html files', () => {
+    const pastebin = pastebinState.get().getOrDie('Could not get pastebin from state');
+
+    pasteFiles(pastebin, [
+      createFileFromString('a.txt', 123, 'Hello world!', 'text/plain'),
+      createFileFromString('a.html', 123, '<b>Hello world!</b>', 'text/html')
+    ]);
+
+    const dataTransfer = pasteState.get().getOrDie('Could not get dataTransfer from state');
+
+    assert.equal(dataTransfer.items.length, 2);
+    assertFileItem(dataTransfer.items[0], { type: 'text/plain', name: 'a.txt', data: 'Hello world!' });
+    assertFileItem(dataTransfer.items[1], { type: 'text/html', name: 'a.html', data: '<b>Hello world!</b>' });
+  });
 
-  Insert.append(SugarBody.body(), pastebin);
+  it('Paste using dataTransfer mutator', () => {
+    const pastebin = pastebinState.get().getOrDie('Could not get pastebin from state');
 
-  const cutUnbinder = DomEvent.bind(pastebin, 'cut', (evt) => {
-    const dataTransfer = evt.raw.clipboardData;
-    dataTransfer.clearData();
-    dataTransfer.setData('text/plain', 'cut-data');
+    pasteDataTransfer(pastebin, (dataTransfer) => {
+      dataTransfer.items.add(createFileFromString('a.txt', 123, 'Hello world!', 'text/plain'));
```

Comment:
same problem as above change the `text` doesn't make the test fail (the line below this works as expected)